src/main/webapp/itempagescript.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Retrieves ItemPageData and forms page using other functions.
+ */
+async function loadItemPage() {
+  const itemId = getItemId();
+  console.log(itemId);
+  if (itemId != null) {
+    fetch(`/itempagedata?itemId=${itemId}`)
+      .then((response) => response.json())
+      .then((ItemPageData) => {
+        createSelectedItemCard(ItemPageData.item);
+        getItemPageComments(ItemPageData.comments);
+      });
+  } else {
+    console.log('ItemId is null');
+  }
+}
+
+/**
+ * Makes Entertainment Item into card
+ */
+function createSelectedItemCard(entertainmentItem) {
+  const card = $('<div class="card bg-light"></div>');
+  card.append(
+    $('<img class="card-img-top" src="' + entertainmentItem.imageUrl + '">'));
+  const cardBody = $('<div class="card-body"></div>');
+  cardBody.append(
+    $('<h5 class="card-title">' + entertainmentItem.title + '</h5>'));
+  cardBody.append(
+    $('<p class="card-text">' + entertainmentItem.description + '</p>'));
+
+  card.append(cardBody);
+  const itemContainer = $('#item-container');
+  itemContainer.append(card);
+}
+
+/**
+ * Gets itemId from current URL
+ */
+function getItemId() {
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams.get('itemId')) {
+    return urlParams.get('itemId');
+  }
```

Comment:
It seems like you are not returning anything if urlParams.get('itemId') is null...?

You could just always `return urlParams.get('itemId');` without the null check.